File: LeetCode/01748. Sum of Unique Elements.py
class Solution:
    def sumOfUnique(self, nums):
        if len(nums) == 1:
            return nums[0]
        nums = sorted(nums)
        sum = 0
        for index, num in enumerate(nums):
            if num not in nums[index+1:] and num != nums[index-1]:
                sum += nums[index]
        return sum


# another solution
from collections import defaultdict

class Solution:
    def sumOfUnique(self, nums: List[int]) -> int:
        counts = defaultdict(int)
        for num in nums:
            counts[num] += 1

        return sum(key for key in counts if counts[key] == 1)

File: LeetCode/01748. Sum of Unique Elements.java
class Solution {
  public int sumOfUnique(int[] nums) {
    Map<Integer, Integer> counts = new HashMap<>();
    for (int num : nums) {
        counts.put(num, counts.getOrDefault(num, 0) + 1);
    }

    return counts.keySet().stream()
            .filter(key -> counts.get(key) == 1)
            .mapToInt(Integer::intValue)
            .sum();
    }
}

// another solution
class Solution {
    public int sumOfUnique(int[] nums) {
        Set<Integer> numsSet = new HashSet<Integer>();
        List<Integer> duplicates = new ArrayList<Integer>();
        int sum = 0;
        for (int num : nums) {
            if (!numsSet.add(num)) {
                duplicates.add(num);
            } 
        }

        for (int num : numsSet) {
            if (!duplicates.contains(num)) {
                sum += num;
            } 
        }
        return sum;
    }
}
